rita/src/rita_client/dashboard/network_endpoints.rs

```diff
@@ -370,3 +372,102 @@ pub fn remote_logging_level(path: Path<String>) -> Box<Future<Item = HttpRespons
 
     return Box::new(future::ok(HttpResponse::Ok().json(())));
 }
+
+pub fn add_exits(
+    new_exits: Json<HashMap<String, ExitServer>>,
+) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    debug!("/exits POST hit with {:?}", new_exits);
+    let exits = &mut SETTING.get_exit_client_mut().exits;
+    exits.extend(new_exits.into_inner());
+
+    Box::new(future::ok(HttpResponse::Ok().json(exits.clone())))
+}
+
+pub fn exits_sync(
+    list_url_json: Json<HashMap<String, String>>,
+) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    debug!("/exits/sync hit with {:?}", list_url_json);
+
+    let list_url = match list_url_json.get("url") {
+        Some(url) if url.starts_with("https://") => url,
+        Some(_unsafe_url) => {
+            let mut ret = HashMap::new();
+            ret.insert(
+                "error".to_owned(),
+                "Attempted to use a non-HTTPS url".to_owned(),
+            );
+            return Box::new(future::ok(
+                HttpResponse::new(StatusCode::BAD_REQUEST)
+                    .into_builder()
+                    .json(ret),
+            ));
+        }
+        None => {
+            let mut ret = HashMap::new();
+
+            ret.insert(
+                "error".to_owned(),
+                "Could not find a \"url\" key in supplied JSON".to_owned(),
+            );
+            return Box::new(future::ok(
+                HttpResponse::new(StatusCode::BAD_REQUEST)
+                    .into_builder()
+                    .json(ret),
+            ));
+        }
+    };
+
+    let client = reqwest::Client::builder()
+        .timeout(Duration::from_secs(5))
+        .build()
+        .unwrap();
+
+    let new_exits: HashMap<String, ExitServer> = match client.get(list_url).send() {
+        Ok(mut response) => match response.json() {
+            Ok(deserialized) => deserialized,
+            Err(e) => {
+                let mut ret = HashMap::<String, String>::new();
+
+                error!(
+                    "Could not deserialize exit list at {:?} because of error: {:?}",
+                    list_url, e
+                );
+                ret.insert(
+                    "error".to_owned(),
+                    format!("Could not deserialize exit list at URL {:?}", list_url),
+                );
+
+                return Box::new(future::ok(
+                    HttpResponse::new(StatusCode::BAD_REQUEST)
+                        .into_builder()
+                        .json(ret),
+                ));
+            }
+        },
+        Err(e) => {
+            let mut ret = HashMap::<String, String>::new();
```

Comment:
I think in this case rust can *probably* will infer type of HashMap, and other similiar cases.

```suggestion
            let mut ret = HashMap::new();
```

Reply:
That's true, needed this for a dummy version, but that's not the case anymore